### PR TITLE
[upstream] Added hyperv perfomance stage

### PR DESCRIPTION
**Pull request by asofron**

### CODE REVIEW

**Comment by ader1990** [comment]
Start the commit message with a tag for the corresponding pipeline, in your case [upstream], followed by a past tense verb, in your case "Added hyperv perfomance stage"

File: linux_pipeline/Jenkinsfile_upstream_kernel [hunk 1]
```diff
@@ -16,8 +16,7 @@ def reportStageStatus(stageName, stageStatus) {
                                     variable: 'PERF_DB_CONFIG')]) {
         dir('kernel_version_report' + env.BUILD_NUMBER + env.BRANCH_NAME) {
               unstash 'kernel_version_ini'
-              sh returnStatus: true, script: '''
-                  #!/bin/bash
```
Comment (ader1990) [inline linux_pipeline/Jenkinsfile_upstream_kernel]: why do you need this change?

File: linux_pipeline/Jenkinsfile_upstream_kernel [hunk 2]
```diff
@@ -33,15 +32,26 @@ def reportStageStatus(stageName, stageStatus) {
         }
     }
 }
+def GetPerfServer(xmlName) {
+    switch(xmlName) {
```
Comment (ader1990) [inline linux_pipeline/Jenkinsfile_upstream_kernel]: what happens if xml name is in another case?

**Comment by ader1990** [comment]
You need to have only one commit per feature/bug/major change, please squash the "minor fix" commits to the main one.

File: linux_pipeline/Jenkinsfile_upstream_kernel [hunk 3]
```diff
@@ -33,15 +32,26 @@ def reportStageStatus(stageName, stageStatus) {
...
+    }
 
 pipeline {
   parameters {
     choice(choices: 'git://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git\nhttps://git.kernel.org/pub/scm/linux/kernel/git/davem/net-next.git', description: 'Kernel tree repo', name: 'KERNEL_GIT_URL')
     string(defaultValue: "master", description: 'Branch to be built', name: 'KERNEL_GIT_BRANCH')
     choice(choices: 'Ubuntu_16.04.3\nCentOS_7.4', description: 'Distro version.', name: 'DISTRO_VERSION')
     choice(choices: "kernel_pipeline_bvt.xml\nkernel_pipeline_fvt.xml\ntest_kernel_pipeline.xml", description: 'Which tests should LISA run', name: 'LISA_TEST_XML')
+    choice(defaultValue: "TCP_SRIOV_v4.xml" choices: "TCP_SRIOV_v4.xml\nUDP_SRIOV_v4.xml\nFIO_RAID0_Pipeline.xml", description: 'Which xml should LISA run for performance test', name: 'LISA_PERF_TEST_XML')
+    string(defaultValue: "", description: 'Options for perf test.', name: 'LISA_PERF_OPTIONS')
```
Comment (papagalu) [inline linux_pipeline/Jenkinsfile_upstream_kernel]: what is LISA_PERF_OPTIONS, what is it supposed to look like? maybe give an example

File: linux_pipeline/Jenkinsfile_upstream_kernel [hunk 4]
```diff
@@ -33,15 +32,26 @@ def reportStageStatus(stageName, stageStatus) {
...
 }
+def GetPerfServer(xmlName) {
+    switch(xmlName) {
+        case ~/TCP.*/:
+            return "net_perf"
+        case ~/UDP.*/:
+            return "net_perf"
+        case ~/FIO.*/:
+            return "stor_perf"
+    }
```
Comment (papagalu) [inline linux_pipeline/Jenkinsfile_upstream_kernel]: missing }

File: linux_pipeline/Jenkinsfile_upstream_kernel [hunk 5]
```diff
@@ -33,15 +32,26 @@ def reportStageStatus(stageName, stageStatus) {
...
 
 pipeline {
   parameters {
     choice(choices: 'git://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git\nhttps://git.kernel.org/pub/scm/linux/kernel/git/davem/net-next.git', description: 'Kernel tree repo', name: 'KERNEL_GIT_URL')
     string(defaultValue: "master", description: 'Branch to be built', name: 'KERNEL_GIT_BRANCH')
     choice(choices: 'Ubuntu_16.04.3\nCentOS_7.4', description: 'Distro version.', name: 'DISTRO_VERSION')
     choice(choices: "kernel_pipeline_bvt.xml\nkernel_pipeline_fvt.xml\ntest_kernel_pipeline.xml", description: 'Which tests should LISA run', name: 'LISA_TEST_XML')
+    choice(defaultValue: "TCP_SRIOV_v4.xml" choices: "TCP_SRIOV_v4.xml\nUDP_SRIOV_v4.xml\nFIO_RAID0_Pipeline.xml", description: 'Which xml should LISA run for performance test', name: 'LISA_PERF_TEST_XML')
```
Comment (papagalu) [inline linux_pipeline/Jenkinsfile_upstream_kernel]: no defaultValue for choices, the first choice will be chosen by default

File: linux_pipeline/Jenkinsfile_upstream_kernel (hunk 3 above)
Comment (asofron) [reply linux_pipeline/Jenkinsfile_upstream_kernel]: After talking with @chvalean , I will remove LISA_PERF_OPTIONS as we are doing standard runs with no modifications to performance variables .

File: linux_pipeline/Jenkinsfile_upstream_kernel [hunk 6]
```diff
@@ -16,8 +16,7 @@ def reportStageStatus(stageName, stageStatus) {
                                     variable: 'PERF_DB_CONFIG')]) {
         dir('kernel_version_report' + env.BUILD_NUMBER + env.BRANCH_NAME) {
               unstash 'kernel_version_ini'
-              sh returnStatus: true, script: '''
-                  #!/bin/bash
+               sh returnStatus: true, script: '''
                   KERNEL_SOURCE="${KERNEL_GIT_URL##*/}"
```
Comment (papagalu) [inline linux_pipeline/Jenkinsfile_upstream_kernel]: indentation should of 4

File: linux_pipeline/Jenkinsfile_upstream_kernel [hunk 7]
```diff
@@ -17,7 +17,6 @@ def reportStageStatus(stageName, stageStatus) {
         dir('kernel_version_report' + env.BUILD_NUMBER + env.BRANCH_NAME) {
               unstash 'kernel_version_ini'
               sh returnStatus: true, script: '''
-                  #!/bin/bash
```
Comment (papagalu) [inline linux_pipeline/Jenkinsfile_upstream_kernel]: why do you remove the shebang?

File: linux_pipeline/Jenkinsfile_upstream_kernel [hunk 8]
```diff
@@ -33,15 +32,27 @@ def reportStageStatus(stageName, stageStatus) {
         }
     }
 }
-
```
Comment (papagalu) [inline linux_pipeline/Jenkinsfile_upstream_kernel]: let one line between methods

**Comment by ader1990** [comment]
Leave a space here:[upstream]Added

File: linux_pipeline/Jenkinsfile_upstream_kernel [hunk 9]
```diff
@@ -681,22 +692,43 @@ pipeline {
...
-        }
-        stage('validation_perf_hyperv') {
-          when {
-            beforeAgent true
-            expression { params.ENABLED_STAGES.contains('validation_perf_hyperv') }
-          }
-          agent {
-            node {
-              label 'meta_slave'
+                stage('validation_perf_hyperv') {
```
Comment (ader1990) [inline linux_pipeline/Jenkinsfile_upstream_kernel]: don t change the indentation, as you cannot see your actual changes

File: linux_pipeline/Jenkinsfile_upstream_kernel [hunk 10]
```diff
@@ -681,22 +692,43 @@ pipeline {
...
+                    }
+                }
             }
-          }
-          steps {
-            echo "NOOP Hyper-V Performance test."
-          }
         }
-      }
+    }
```
Comment (ader1990) [inline linux_pipeline/Jenkinsfile_upstream_kernel]: you have wrong number of matching start/end {}

File: linux_pipeline/Jenkinsfile_upstream_kernel [hunk 11]
```diff
@@ -423,8 +438,6 @@ pipeline {
                                        string(credentialsId: 'AZURE_SAS', variable: 'AZURE_SAS'),
                                        string(credentialsId: 'UPSTREAM_AZURE_STORAGE_URL', variable: 'UPSTREAM_AZURE_STORAGE_URL'),
                                        string(credentialsId: 'LISA_TEST_DEPENDENCIES', variable: 'LISA_TEST_DEPENDENCIES'),
-                                       string(credentialsId: 'DB_CONFIG_UPSTREAM', variable: 'DB_CONFIG_UPSTREAM'),
```
Comment (ader1990) [inline linux_pipeline/Jenkinsfile_upstream_kernel]: leave this as they are, they are not the purpose of this commit.
if you prefer to change, you need to create a special commit/PR for the changes

File: linux_pipeline/Jenkinsfile_upstream_kernel [hunk 12]
```diff
@@ -447,8 +460,6 @@ pipeline {
                         -LisaTestDependencies "${env:LISA_TEST_DEPENDENCIES}"
                         -PipelineName "pipeline-linux-upstream-validation/${env:BRANCH_NAME}"
                         -DBConfigPath "${env:DBConfigPath}"
-                        -LisaLogDBConfigPath "${env:DB_CONFIG_UPSTREAM}"
```
Comment (ader1990) [inline linux_pipeline/Jenkinsfile_upstream_kernel]: ditto (same as above)

File: linux_pipeline/Jenkinsfile_upstream_kernel [hunk 13]
```diff
@@ -681,22 +692,44 @@ pipeline {
...
-        }
-        stage('validation_perf_hyperv') {
-          when {
-            beforeAgent true
-            expression { params.ENABLED_STAGES.contains('validation_perf_hyperv') }
-          }
-          agent {
-            node {
-              label 'meta_slave'
+        }
```
Comment (ader1990) [inline linux_pipeline/Jenkinsfile_upstream_kernel]: do not change the indentation if not needed

Comment (asofron) [reply linux_pipeline/Jenkinsfile_upstream_kernel]: OK .I`ll try not to touch indentation

File: linux_pipeline/Jenkinsfile_upstream_kernel [hunk 14]
```diff
@@ -681,22 +695,42 @@ pipeline {
...
-              label 'meta_slave'
+        }  
+                stage('validation_perf_hyperv') {
+                    when {
+                        beforeAgent true
+                        expression { params.ENABLED_STAGES.contains('validation_perf_hyperv') }
+                    }
+                    agent {
+                        node {
+                            label GetPerfServer(LISA_PERF_TEST_XML)
```
Comment (ader1990) [inline linux_pipeline/Jenkinsfile_upstream_kernel]: use env.LISA_PERF_TEST_XML

File: linux_pipeline/Jenkinsfile_upstream_kernel [hunk 15]
```diff
@@ -448,7 +461,6 @@ pipeline {
                         -PipelineName "pipeline-linux-upstream-validation/${env:BRANCH_NAME}"
                         -DBConfigPath "${env:DBConfigPath}"
                         -LisaLogDBConfigPath "${env:DB_CONFIG_UPSTREAM}"
-                        -LogsPath "${env:LISA_UPSTREAM_LOG_DESTINATION}"
```
Comment (ader1990) [inline linux_pipeline/Jenkinsfile_upstream_kernel]: leave this line as it was

File: linux_pipeline/Jenkinsfile_upstream_kernel [hunk 16]
```diff
@@ -689,14 +701,34 @@ pipeline {
...
           agent {
             node {
-              label 'meta_slave'
+              label GetPerfServer(env.LISA_PERF_TEST_XML)
             }
           }
-          steps {
-            echo "NOOP Hyper-V Performance test."
-          }
+           steps {
```
Comment (ader1990) [inline linux_pipeline/Jenkinsfile_upstream_kernel]: use a multiple of 2 spaces indentation, to have the same indentation used before.

Comment (ader1990) [reply linux_pipeline/Jenkinsfile_upstream_kernel]: you have one space at steps

File: linux_pipeline/Jenkinsfile_upstream_kernel [hunk 17]
```diff
@@ -689,14 +701,34 @@ pipeline {
...
+            withCredentials(bindings: [string(credentialsId: 'LISA_IMAGES_SHARE_URL', variable: 'LISA_IMAGES_SHARE_URL'),
```
Comment (ader1990) [inline linux_pipeline/Jenkinsfile_upstream_kernel]: ditto

File: linux_pipeline/Jenkinsfile_upstream_kernel [hunk 18]
```diff
@@ -689,14 +701,34 @@ pipeline {
...
+            string(credentialsId: 'LISA_TEST_DEPENDENCIES', variable: 'LISA_TEST_DEPENDENCIES')]) {
+            echo 'Running LISA...'
+            unstash "${env.KERNEL_ARTIFACTS_PATH}"
+            PowerShellWrapper('''
+            & ".\\scripts\\lis_hyperv_platform\\main_perf.ps1"
+            -JobId "${env:BUILD_NAME}${env:BUILD_NUMBER}${env:BRANCH_NAME}-perf"
```
Comment (ader1990) [inline linux_pipeline/Jenkinsfile_upstream_kernel]: ditto

File: linux_pipeline/Jenkinsfile_upstream_kernel [hunk 19]
```diff
@@ -689,14 +702,34 @@ pipeline {
...
           agent {
             node {
-              label 'meta_slave'
+              label GetPerfServer(env.LISA_PERF_TEST_XML)
             }
           }
           steps {
-            echo "NOOP Hyper-V Performance test."
+            withCredentials(bindings: [string(credentialsId: 'LISA_IMAGES_SHARE_URL', variable: 'LISA_IMAGES_SHARE_URL'),
+            string(credentialsId: 'LISA_TEST_DEPENDENCIES', variable: 'LISA_TEST_DEPENDENCIES')]) {
```
Comment (ader1990) [inline linux_pipeline/Jenkinsfile_upstream_kernel]: add indentation

File: linux_pipeline/Jenkinsfile_upstream_kernel [hunk 20]
```diff
@@ -689,14 +702,34 @@ pipeline {
...
+            echo 'Running LISA...'
+            unstash "${env.KERNEL_ARTIFACTS_PATH}"
+            PowerShellWrapper('''
```
Comment (ader1990) [inline linux_pipeline/Jenkinsfile_upstream_kernel]: add a new line

File: linux_pipeline/Jenkinsfile_upstream_kernel [hunk 21]
```diff
@@ -689,14 +702,34 @@ pipeline {
...
+              -JobId "${env:BUILD_NAME}${env:BUILD_NUMBER}${env:BRANCH_NAME}-perf"
+              -InstanceName "${env:BUILD_NAME}${env:BUILD_NUMBER}${env:BRANCH_NAME}-p"
+              -VHDType "${env:DISTRO_VERSION}.ToLower().Split('_')[0]" -WorkingDirectory "C:\\workspace"
+              -OSVersion "${env:DISTRO_VERSION}.Split('_')[1]" -LISAManageVMS:$true
+              -LISAImagesShareUrl "${env:LISA_IMAGES_SHARE_URL}" -XmlTest "${env:LISA_PERF_TEST_XML}"
+              -LisaTestDependencies "${env:LISA_TEST_DEPENDENCIES}"
+              -LocalKernelFolder "scripts\\package_building\\${env:BUILD_NUMBER}-${env:BRANCH_NAME}-${env:KERNEL_ARTIFACTS_PATH}\\**\\deb"
+            ''')
+            echo 'Finished running validation_perf_hyperv ...'
+              }
```
Comment (ader1990) [inline linux_pipeline/Jenkinsfile_upstream_kernel]: remove indentation

File: linux_pipeline/Jenkinsfile_upstream_kernel [hunk 22]
```diff
@@ -689,14 +702,34 @@ pipeline {
...
+            }
+          post {
```
Comment (ader1990) [inline linux_pipeline/Jenkinsfile_upstream_kernel]: ditto

File: linux_pipeline/Jenkinsfile_upstream_kernel [hunk 23]
```diff
@@ -689,14 +702,34 @@ pipeline {
...
+            always {
+              archiveArtifacts "${BUILD_NAME}${BUILD_NUMBER}${BRANCH_NAME}-perf\\lis-test\\WS2012R2\\lisa\\TestResults\\**\\*"
+              }
```
Comment (ader1990) [inline linux_pipeline/Jenkinsfile_upstream_kernel]: ditto

File: linux_pipeline/Jenkinsfile_upstream_kernel [hunk 24]
```diff
@@ -689,11 +702,32 @@ pipeline {
...
             node {
-              label 'meta_slave'
+              label GetPerfServer(env.LISA_PERF_TEST_XML)
             }
           }
           steps {
-            echo "NOOP Hyper-V Performance test."
+            withCredentials(bindings: [string(credentialsId: 'LISA_IMAGES_SHARE_URL', variable: 'LISA_IMAGES_SHARE_URL'),
+                                       string(credentialsId: 'LISA_TEST_DEPENDENCIES', variable: 'LISA_TEST_DEPENDENCIES')]) {
+              echo 'Running LISA...'
```
Comment (chvalean) [inline linux_pipeline/Jenkinsfile_upstream_kernel]: please change to "Running LISA - performance..."

File: linux_pipeline/Jenkinsfile_upstream_kernel [hunk 25]
```diff
@@ -689,11 +702,32 @@ pipeline {
           }
           agent {
             node {
-              label 'meta_slave'
+              label GetPerfServer(env.LISA_PERF_TEST_XML)
             }
           }
           steps {
-            echo "NOOP Hyper-V Performance test."
+            withCredentials(bindings: [string(credentialsId: 'LISA_IMAGES_SHARE_URL', variable: 'LISA_IMAGES_SHARE_URL'),
```
Comment (chvalean) [inline linux_pipeline/Jenkinsfile_upstream_kernel]: the hyper-v perf stage must be conditioned to run only for ubuntu

**Pull request merged**